features/dd-sdk-android-ndk/src/main/cpp/datadog-native-lib.cpp

```diff
@@ -43,9 +49,10 @@ void write_crash_report(int signum,
     static const std::string crash_log_filename = "crash_log";
 
     // sync everything
-    pthread_mutex_lock(&handler_mutex);
     if (tracking_consent != tracking_consent_granted) {
-        pthread_mutex_unlock(&handler_mutex);
+        return;
+    }
+    if(pthread_mutex_trylock(&handler_mutex)!=0){
```

Comment:
```suggestion
    if(pthread_mutex_trylock(&handler_mutex) != 0){
```

Reply:
I usually apply ktlintFormat at the end of my PR but of course for C code doesn't work :)